qa/workunits/ceph-helpers.sh

```diff
@@ -197,26 +197,30 @@ function kill_daemons() {
     local dir=$1
     local signal=${2:-TERM}
     local name_prefix=$3 # optional, osd, mon, osd.1
-    local delays=${4:-0 0 1 1 1 2 3 5 5 5 10 10 20 60 60 60 120}
+    local delays=${4:-0.1 0.2 1 1 1 2 3 5 5 5 10 10 20 60 60 60 120}
 
     local status=0
+    pids=""
     for pidfile in $(find $dir 2>/dev/null | grep $name_prefix'[^/]*\.pid') ; do
-        pid=$(cat $pidfile)
+        (pid=$(cat $pidfile)
         local send_signal=$signal
-        local kill_complete=false
+        local exit_code=1
         for try in $delays ; do
-            sleep $try
             if kill -$send_signal $pid 2> /dev/null ; then
-                kill_complete=false
+	        exit_code=1
             else
-                kill_complete=true
+	        exit_code=0
                 break
             fi
             send_signal=0
-        done
-        if ! $kill_complete ; then
-            status=1
-        fi
+            sleep $try
+        done; exit $exit_code) &
+	pids="$pids $!"
+    done
+    for pid in $pids; do
+	if ! wait $pid; then
```

Comment:
I like this optimization :-) It needs to be tested in test_kill_daemons (the race mentionned above can be faked by making it so the first daemon in the list blocs and the second one terminates quickly).